tests/pytest/test_binary_value.py

```diff
@@ -8,6 +8,39 @@
 TRUNCATION_MATCH = r"\d+-bit value requested, truncating value"
 
 
+def test_init_zero_length():
+    bin1 = BinaryValue(value=0, n_bits=0, binaryRepresentation=BinaryRepresentation.UNSIGNED)
+    assert bin1._str == ""
+    assert bin1.binstr == ""
+    assert bin1.integer == 0
+
+    bin2 = BinaryValue(value=0, n_bits=0,
+                       binaryRepresentation=BinaryRepresentation.TWOS_COMPLEMENT)
+    assert bin2._str == ""
+    assert bin2.binstr == ""
+    assert bin2.integer == 0
+
+    bin3 = BinaryValue(value=0, n_bits=0,
+                       binaryRepresentation=BinaryRepresentation.SIGNED_MAGNITUDE)
+    assert bin3._str == ""
+    assert bin3.binstr == ""
+    assert bin3.integer == 0
+
+    # Whatever value is set to a zero bit long BinaryValue, it should read 0
+    with pytest.warns(RuntimeWarning, match=TRUNCATION_MATCH):
+        bin4 = BinaryValue(value=10, n_bits=0,
+                           binaryRepresentation=BinaryRepresentation.SIGNED_MAGNITUDE)
+    assert bin4._str == ""
+    assert bin4.binstr == ""
+    assert bin4.integer == 0
+
+    with pytest.warns(RuntimeWarning, match=TRUNCATION_MATCH):
+        bin4 <= 5
```

Comment:
Are you sure `<=` is legal on `BinaryValue`?

Reply:
No not sure but it works. I can remove this test (was more a test for me)

Reply:
It should be legal (line 709 of binary.py and below)

>     def __setitem__(self, key, val):
        """BinaryValue uses Verilog/VHDL style slices as opposed to Python style."""
        if not isinstance(val, str) and not isinstance(val, int):

Reply:
Should be legal:

> `    def __le__(self, other):
        self.assign(other)`
( binary.py :473)